### PR TITLE
Updates to Url types for Pydantic V2

### PR DESCRIPTION
**Pull request by tpdorsey**
Review and update Url types, removing outdated information and updating examples.

## Related issue number

Closes PYD-171

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @davidhewitt

### CODE REVIEW

**Comment by tpdorsey** [comment]
please review

File: docs/usage/types/urls.md [hunk 1]
```diff
@@ -2,56 +2,50 @@
...
-    - `allowed_schemes: Optional[Set[str]] = None`
-
-!!! warning
-    In V1.10.0 and v1.10.1 `stricturl` also took an optional `quote_plus` argument and URL components were percent
-    encoded in some cases. This feature was removed in v1.10.2, see
-    [#4470](https://github.com/pydantic/pydantic/pull/4470) for explanation and more details.
-
-The above types (which all inherit from `AnyUrl`) will attempt to give descriptive errors when invalid URLs are
-provided:
+- [`AnyUrl`][pydantic.networks.AnyUrl]: any scheme allowed, top-level domain (TLD) not required, host required.
```
Comment (davidhewitt) [inline docs/usage/types/urls.md:5]: I wonder if it makes sense to tabulate these and maybe even generate the table programmatically?

File: docs/usage/types/urls.md [hunk 2]
```diff
@@ -2,56 +2,50 @@
...
+    - `postgres`
+    - `postgresql`
+    - `postgresql+asyncpg`
+    - `postgresql+pg8000`
+    - `postgresql+psycopg`
+    - `postgresql+psycopg2`
+    - `postgresql+psycopg2cffi`
+    - `postgresql+py-postgresql`
+    - `postgresql+pygresql`
+- [`MySQLDsn`][pydantic.networks.MySQLDsn]: scheme `mysql`, user info required, TLD not required, host required.
```
Comment (davidhewitt) [inline docs/usage/types/urls.md:20]: I wonder if we want to link to definitions of all of these, where possible? I assume it's easy enough for users to search for these though having links to e.g. mysql docs may be a nice touch? I suppose it's also a lot of effort to collect and maintain external links, so maybe not worthwhile.

**Review by davidhewitt — APPROVED** [review]
👍 lgtm, just some offhand thoughts which you can freely dismiss 😁

**Review by Kludex — APPROVED** [review]

File: docs/usage/types/urls.md (hunk 1 above)
Comment (tpdorsey) [reply docs/usage/types/urls.md:5]: This is a good callout. I'll look into it today.

**Pull request merged**